### DISCUSSION TITLE
Usage possibility with inference done before hand.

### DISCUSSION THREAD

**Discussion by Nachimak28** [Project-MONAI/MONAILabel#856] [Q&A] [answered]
Hi there
First of all a big thank you to the monai team for creating these awesome integrations for the tools. 

As recommended in other discussion topics, its better to run the monailabel server on a GPU. My use case is such that my team has to run the server on a GCP cloud VM and make it accessible to a team of radiologists doing annotations remotely. We love the semi-autonomous annotations given it reduces the annotation time significantly but we wish to make it non interactive. We do have the cloud provision but cannot afford to keep a GPU running all day (for about a month) while the annotators complete the annotation. We could keep a CPU running continuously for some time though and wish to deploy the monailabel server in it. 

I wanted to ask whether we can do the inference in bulk beforehand and simply place the labels in a directory from where the monailabel server can load the annotations without having a model to infer everytime a user needs the pre-generated annotations to modify if necessary. Our project's scope does not include active learning and training to happen and want the monailabel server to merely help in annotation with quick pre-generated inference loading.

We do have the flexibility of using any of the tools: 3D slicer or OHIF based on whichever tools supports the above mentioned issue.

**Comment by diazandr3s** [comment, marked as answer]
Hi @Nachimak28,

Thanks for openning this dicussion, this is a good question.

You could run batch inference right after you start the server. For this, make a POST call for batch inference using the browser or a bash script.

Batch inference results save the predictions in the labels/original folder.

Here it is a bash script to do exactly that: https://github.com/Project-MONAI/MONAILabel/blob/bratsSegmentation/sample-apps/radiology/start_server.sh

Hope that helps,

**Reply by Nachimak28** [reply]
@diazandr3s Thank you so much for the prompt response. This would be really helpful. One question though, would this work with studies loaded in orthanc too or just for studies loaded from the directory?

**Reply by diazandr3s** [reply]
Very good question. I'm not sure whether the predictions are saved in the Orthanc (DICOM Web server) after performing batch inference. 
We use "original" as a tag to store predictions when working from a directory (https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/datastore/local.py#L139)
When using dicom, I don't think this is supported yet. You can check the datastore classes here: https://github.com/Project-MONAI/MONAILabel/tree/main/monailabel/datastore
@SachidanandAlle what do you think?

**Reply by Nachimak28** [reply]
No worries, I'll try it out on my end and as I said, we're flexible in terms of tools. If batch inference for orthanc is not possible, then i'll simply use 3D slicer, convert dicoms to nifti format and load them from the directory.
Thanks again for the quick response.
